### PR TITLE
[JUJU-2586] Fix LP19988587 Can't switch from edge channel to stable channel

### PR DESCRIPTION
**Pull request by hmlanigan**
Allow an application's channel to be updated when refreshing even if the charm revision is the same so no other change is needed. This fixes a regression in behavior between charm hub and charm store charms. Because it's a regression, only a small change was required.

## QA steps

Today, all risk levels of the latest track for the ubuntu charm are the same. Makes a good test candidate.

```sh
$ juju deploy ubuntu
$ juju refresh ubuntu --channel edge
charm "ubuntu": already up-to-date. Note: all future refreshes will now use channel "edge"
```
## Bug reference

https://bugs.launchpad.net/juju/+bug/1988587

### CODE REVIEW

**Review by cderici — APPROVED** [review]
LGTM. QA steps confirmed. Played with it a little bit, seems to be working well :+1:  Also fwiw the newly added tests are passing for me.

File: cmd/juju/application/refresh.go [hunk 1]
```diff
@@ -439,21 +440,20 @@ func (c *refreshCommand) Run(ctx *cmd.Context) error {
 			channel = fmt.Sprintf(" in channel %s", charmID.Origin.Channel.String())
 		}
 		ctx.Infof("Added %s charm %q, revision %d%s, to the model", charmOrigin.Source, curl.Name, curl.Revision, channel)
-	} else if errors.Is(err, refresher.ErrAlreadyUpToDate) {
-		if len(c.Resources) == 0 {
```
Comment (hmlanigan) [inline cmd/juju/application/refresh.go:443]: This logic has been broken. Need to fix.

**Review by cderici — APPROVED** [review]
Yup seems to be working well now 👍 LGTM

**Pull request merged**